src/node_context_data.h

```diff
@@ -21,26 +24,62 @@ namespace node {
 #define NODE_CONTEXT_ALLOW_WASM_CODE_GENERATION_INDEX 34
 #endif
 
-#ifndef NODE_CONTEXT_TAG
-#define NODE_CONTEXT_TAG 35
-#endif
-
 #ifndef NODE_BINDING_LIST
-#define NODE_BINDING_LIST_INDEX 36
+#define NODE_BINDING_LIST_INDEX 35
 #endif
 
 #ifndef NODE_CONTEXT_ALLOW_CODE_GENERATION_FROM_STRINGS_INDEX
-#define NODE_CONTEXT_ALLOW_CODE_GENERATION_FROM_STRINGS_INDEX 37
+#define NODE_CONTEXT_ALLOW_CODE_GENERATION_FROM_STRINGS_INDEX 36
+#endif
+
+// NODE_CONTEXT_TAG must be greater than any embedder indexes so that a single
+// check on the number of embedder data fields can assure the presence of all
+// embedder indexes.
+#ifndef NODE_CONTEXT_TAG
+#define NODE_CONTEXT_TAG 37
 #endif
 
 enum ContextEmbedderIndex {
   kEnvironment = NODE_CONTEXT_EMBEDDER_DATA_INDEX,
   kSandboxObject = NODE_CONTEXT_SANDBOX_OBJECT_INDEX,
   kAllowWasmCodeGeneration = NODE_CONTEXT_ALLOW_WASM_CODE_GENERATION_INDEX,
-  kContextTag = NODE_CONTEXT_TAG,
   kBindingListIndex = NODE_BINDING_LIST_INDEX,
   kAllowCodeGenerationFromStrings =
-      NODE_CONTEXT_ALLOW_CODE_GENERATION_FROM_STRINGS_INDEX
+      NODE_CONTEXT_ALLOW_CODE_GENERATION_FROM_STRINGS_INDEX,
+  kContextTag = NODE_CONTEXT_TAG,
+};
+
+class ContextEmbedderTag {
```

Comment:
Does this have to be a class? It feels more like a namespace to me.

Reply:
We can not declare a private member in a namespace and access it with an inline function, no?

Reply:
Yea, true but is it necessary for these functions to be inlined? These aren't used very often, so maybe not a good candidate for inlining? I think we can declare these functions in this header file and define them in a cc file and define kNodeContextTagPtr and kNodeContextTag in an anonymous namespace in that cc file, wdyt?

Reply:
Environment::GetCurrent() is used in almost every C++ binding so I think it's very important to keep them inlined...